lib/Progress/Logger/BlinkenLogger.php

```diff
@@ -177,9 +180,7 @@ private function drawIterations(Variant $variant, array $specials, $tag): void
         $line = sprintf('%-' . self::INDENT . 's', '#' . $variant->getSubject()->getIndex());
         $nbIterations = $variant->count();
 
-        for ($index = 0; $index < $nbIterations; $index++) {
-            $iteration = $variant->getIteration($index);
-
+        foreach ($variant as $index => $iteration) {
```

Comment:
Are we sure that index is a sequential list here? Maybe use array_values if not

Reply:
Yes, we are sure.
- In `lib/Model/Variant.php` we add values to the private property `Variant::$iterations` using only `$this->iterations[] = XXX`.
- `Variant::$iterations` has `/** @var list<Iteration> */` docblock. It means this array is 0-based sequential list (https://phpstan.org/writing-php-code/phpdoc-types#lists).

Reply:
yes, have some ancient memories about making for loops like that to work around "lists" that had been filtered etc making their array keys non-sequential.